src/components/MapView/DateSelector/index.tsx

```diff
@@ -37,46 +40,34 @@ const Input = forwardRef(
 );
 
 function DateSelector({ availableDates = [], classes }: DateSelectorProps) {
-  const selectedDate = useSelector(selectCurrentDate);
-  const selectedMonth = selectedDate.getMonth();
-  const selectedYear = selectedDate.getFullYear();
   const dispatch = useDispatch();
+  const { startDate: stateStartDate } = useSelector(dateRangeSelector);
+  const stateStartDateYear = moment(stateStartDate).year();
 
-  function isAvailableDate(date: Date) {
-    return Boolean(
-      availableDates.find(dateIt => moment(date).isSame(dateIt, 'day')),
-    );
-  }
+  const [selectedDate, setSelectedDate] = useState(moment(stateStartDate));
+  const selectedMonth = selectedDate.month();
+  const selectedYear = selectedDate.year();
 
-  function isAvailableMonth(month: number) {
-    return Boolean(
-      findAvailableDayInMonth(new Date(selectedYear, month, 1), availableDates),
-    );
-  }
+  useEffect(() => {
+    setSelectedDate(moment(stateStartDate));
+  }, [stateStartDate]);
 
   function updateMonth(month: number) {
-    // Fetching the first available day in the month
-    const date = findAvailableDayInMonth(
-      new Date(selectedYear, month, 1),
-      availableDates,
-    );
-    if (date) {
-      dispatch(updateDate(date));
-    }
+    const { startDate, endDate } = getMonthStartAndEnd(month, selectedYear);
+    dispatch(updateDateRange({ startDate, endDate }));
   }
 
   function incrementYear() {
-    dispatch(updateDate(new Date(selectedYear + 1, selectedMonth, 1)));
+    setSelectedDate(selectedDate.year(selectedYear + 1).clone());
```

Comment:
Is this the best way to do this? I can't remember the `moment` interface, but does `.year` mutate the original object? It probably doesn't matter much in this case, but would `selectedDate.clone().year(selectedYear + 1)` make more sense? 

I assume `.year` mutates the object rather than a copy?